sklearn/model_selection/_classification_threshold.py

```diff
@@ -996,8 +1037,18 @@ def get_metadata_routing(self):
 
     def _get_curve_scorer(self):
         """Get the curve scorer based on the objective metric used."""
-        scoring = check_scoring(self.estimator, scoring=self.scoring)
+        scorer = check_scoring(self.estimator, scoring=self.scoring)
+        # XXX: at the time of writing, there is no very explicit way to check
+        # if a scorer expects thresholded binary classification predictions.
+        # TODO: updates this condition when a better way is available.
+        if scorer._response_method != "predict":
+            raise ValueError(
```

Comment:
This is still the most controversial part. Without it, this PR would be almost merged, right?

Reply:
What do you think of https://github.com/scikit-learn/scikit-learn/pull/29082#discussion_r1613325997 ?